netspresso/benchmarker/v2/benchmarker.py

```diff
@@ -0,0 +1,199 @@
+    ) -> BenchmarkTask:
+        """Benchmark the specified model on the specified device.
+
+        Args:
+            input_model_path (str): The file path where the model is located.
+            target_device_name (DeviceName): Target device name.
+            target_software_version (Union[str, SoftwareVersion], optional): Target software version. Required if target_device_name is one of the Jetson devices.
+            target_hardware_type (Union[str, HardwareType], optional): Hardware type. Acceleration options for processing the model inference.
+            wait_until_done (bool): If True, wait for the conversion result before returning the function.
+                                If False, request the conversion and return the function immediately.
+
+        Raises:
+            e: If an error occurs during the benchmarking of the model.
+
+        Returns:
+            Dict: Model benchmark task dictionary.
+        """
+
+        FileHandler.check_input_model_path(input_model_path)
+
+        self.token_handler.validate_token()
+
+        file_name = "benchmark"
+
+        try:
+            folder_path = Path(input_model_path).parent
+
+            benchmarker_metadata = BenchmarkerMetadata()
+            metadatas = []
+
+            if FileHandler.check_exists(folder_path / f"{file_name}.json"):
+                metadatas = MetadataHandler.load_json(folder_path / f"{file_name}.json")
+                # metadatas.append(asdict(benchmarker_metadata))
+
+            current_credit = auth_client.get_credit(
+                access_token=self.token_handler.tokens.access_token,
+                verify_ssl=self.token_handler.verify_ssl,
+            )
+            check_credit_balance(
+                user_credit=current_credit, service_credit=ServiceCredit.MODEL_BENCHMARK
+            )
+
+            # GET presigned_model_upload_url
+            presigned_url_response = (
+                launcher_client_v2.benchmarker.presigned_model_upload_url(
+                    access_token=self.token_handler.tokens.access_token,
+                    input_model_path=input_model_path,
+                )
+            )
+
+            # UPLOAD model_file
+            launcher_client_v2.benchmarker.upload_model_file(
+                access_token=self.token_handler.tokens.access_token,
+                input_model_path=input_model_path,
+                presigned_upload_url=presigned_url_response.data.presigned_upload_url,
+            )
+
+            # VALIDATE model_file
+            validate_model_response = (
+                launcher_client_v2.benchmarker.validate_model_file(
+                    access_token=self.token_handler.tokens.access_token,
+                    input_model_path=input_model_path,
+                    ai_model_id=presigned_url_response.data.ai_model_id,
+                )
+            )
+
+            # START convert task
+            response = launcher_client_v2.benchmarker.start_task(
```

Comment:
p1
Neet to make process for response status code is not ok(when status code is not 200 things.)

Reply:
Thanks for your feedback. I’ve updated the code as per your suggestions. 👍